srtcore/buffer.cpp

```diff
@@ -161,7 +161,7 @@ void CSndBuffer::addBuffer(const char* data, int len, SRT_MSGCTRL& w_mctrl)
     if (w_srctime == 0)
     {
         HLOGC(dlog.Debug, log << CONID() << "addBuffer: DEFAULT SRCTIME - overriding with current time.");
-        w_srctime = time.us_since_epoch();
+        w_srctime = count_microseconds(time);
```

Comment:
I'm not convinced to the idea of having "a count of microseconds" for an absolute clock. I'd understood if you turn an absolute time into a DURATION since epoch - that would make more sense. And then you wouldn't need a specialization of `count_microseconds` for the time point.

I know this above is simpler, but simplification sometimes makes the instruction misleading, and if so, it's not worth it. And I believe this is the case.

Reply:
It would be `count_microseconds(time.time_since_epoch())` in that case.
Is it what you meant?

Reply:
Like that, but rather `duration_since_epoch`, as what you want to get from here is duration.

Reply:
Resolved